tests/BlazingSQLTest/Runner/runTest.py

```diff
@@ -1707,13 +1181,54 @@ def run_query_log(
             if type(result_gdf) is dask_cudf.core.DataFrame:
                 result_gdf = result_gdf.compute()
 
+            expected_dtypes = result_gdf.dtypes.to_list()
+            pdf1 = (
+                upcast_to_float(result_gdf)
+                .fillna(get_null_constants(result_gdf))
+                .to_pandas()
+            )
+
+            if not isinstance(engine, str):
+                pdf2 = to_pandas_f64_engine(
+                    base_results_gd.resultSet, expected_dtypes
+                )
+                pdf2 = upcast_to_float(pdf2).fillna(get_null_constants(pdf2))
+
+                formatResults(pdf1, pdf2, worder, orderBy)
+            else:
+                if compareResults == "true":
+                        format_pdf(pdf1, worder, orderBy)
+                        pdf2 = get_results(resultFile)
+                        print(pdf2)
+                else:
+                    pdf2 = pd.DataFrame()
+                    formatResults(pdf1, pdf2, worder, orderBy)
+
+            if Settings.execution_mode == ExecutionMode.GENERATOR:
+                results_file_generator(testsWithNulls, filename, engine, pdf2)
+            else:
+                print_query_results(
```

Comment:
Could you please refactor some of this as well? Here the `print_query_results` function also does the validation of the results and query names and a whole bunch more things. The function name is misleading because it sound like this query should "print query results" but its really doing a whole lot more. I would suggest having validation and printing  being separate functions and having them be called sequentially instead of being nested, that way those functions are more reusable and modular.